ARCs/arc-0014.md

```diff
@@ -0,0 +1,67 @@
+---
+ams: 14
+title: Algorand standard for Authentication
+status: Draft
+---
+
+# Algorand Standard for authentication
+
+## Summary
+
+This document introduces the standard for authentication using the algorand accounts.
+
+## Abstract
+
+The goal of this standard is to define authorization process and authentication procedures for communication between web application and backend services or between two backend services. 
+
+## Specification
+
+The key words "**MUST**", "**MUST NOT**", "**REQUIRED**", "**SHALL**", "**SHALL NOT**", "**SHOULD**", "**SHOULD NOT**", "**RECOMMENDED**", "**MAY**", and "**OPTIONAL**" in this document are to be interpreted as described in [RFC-2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+Base64 is the standard described in [RFC-4648](https://www.ietf.org/rfc/rfc4648.txt).
+
+## Request for comments
+
+Please comment or suggest pull request for this document here: [https://github.com/scholtz/AMS/issues](https://github.com/scholtz/AMS/issues)
+
+### Authentication
+
+Authentication is process in which person generates the authorization token for communication with backend service.
+
+Token MUST be generated from Algorand public and private key - from algorand address.
+
+In the first step of the authentication, the message to be signed is generated. The message MUST contain note field with configurable Realm. Realm is identifier of the service. It MAY be the web address of the service for intended purpose.
```

Comment:
I like this idea.

Reply:
maybe define a standard scheme for this so that wallets can let the users know its a "sign in" type request

Reply:
If the note field is to be used for Realm then where one should put nonce (to avoid replay attacks)

Reply:
the note field may still contain a nonce or any other unique data to ensure uniqueness

Reply:
> 

How about this scheme so that we can have a nice , extensible, wallet "sign-in"

auth/v1:j{"realm":"MetaHealth","notice":"You are signing-in to MetaHealth platfrom","nonce":"5aa15e8fa86ecd71455c56871c328070"}

with auth/v1 defined as

interface authInfoV1 {
  realm: string,
  notice?: string,
  nonce?: string,
}


Sample typescript code:
https://gist.github.com/urtho/610fb4e891f665eed2e7955e0fb3e74e

Reply:
i like this idea, even though if user would be able to sign the transaction with replay attack, something is wrong elsewhere.. the uniqueness of the tx is guaranteed by the algorand tx id, but if you guys thing we need to put more information to the message lets do it